### ISSUE TITLE
YOLOv3 support

### ISSUE THREAD

**Issue by wkdhkr** [thtrieu/darkflow#665]
https://pjreddie.com/darknet/yolo/

Can I use YOLOv3 cfg/weights with darkflow correctly?

**Comment by SkullPirateTK** [comment]
Just tested, not working.
`Parsing ./cfg/yolov3.cfg
Layer [shortcut] not implemented`
Hope it will be added soon!

**Comment by ccs1605** [comment]
yes, tested and got same a question!

**Comment by melonetern** [comment]
If you read the official [yolov3-voc.cfg](https://github.com/pjreddie/darknet/blob/master/cfg/yolov3-voc.cfg), you will see the CNN is more big, and that is actually related to the code, obviously, the code need to be rewrote for a new CNN.

**Comment by thtrieu** [comment]
I'll try to do minor refactoring so that the code is easier to read & extend for contributors. I admit after a while not looking at it I'm kind of lost.

**Comment by hf200012** [comment]
YOLO v3 savepb
 Layer [shortcut] not implemented

**Comment by Peddz24** [comment]
Hey @thtrieu,
Is this supported yet? 
Or does this still need to be implemented?
Great work on the repo anyway!

**Comment by eric4note** [comment]
Highly expected.

**Comment by LucasMahieu** [comment]
Here, someone has implemented yolov3 in tensorflow.
https://github.com/mystic123/tensorflow-yolo-v3

**Comment by ambr89** [comment]
Hi @LucasMahieu !
This implementation is only for testing, not for trainig.
I've found this: [https://github.com/raytroop/YOLOv3_tf](https://github.com/raytroop/YOLOv3_tf) but  I haven't tried yet

**Comment by LucasMahieu** [comment]
Hi @ambr89,
Thanks for the link, I will try it.
It would be nicer if Yolov3 was supported in darkflow...

**Comment by ambr89** [comment]
I know, at the moment I'm trying to translate from darknet to darflow the missing layer and new loss function, but it isn't easy !!!

**Comment by LucasMahieu** [comment]
Great !
Which layer is not easy ?
You may need some help ?

**Comment by ambr89** [comment]
At the moment, I'm trying to reproduce loss function.
The upsample layer I hope I did well, but while I haven't finisched I don't know if work.

**Comment by bendidi** [comment]
@ambr89 I can help you with the translation , I have already translated the 3 darknet layers to darkflow (need to  look for the project cause it's been a while) , contact me if you want to colaborate ! 
cheers

**Comment by slothkong** [comment]
@ambr89, @bendidi, @LucasMahieu have you agreed on what is needed to do next? Do you have a list of milestones we could help with?

**Comment by ambr89** [comment]
Hi everybody!
At the moment, I'm using this [project](https://github.com/qqwweee/keras-yolo3)
It isn't darkflow but it's quite similar.

**Comment by LucasMahieu** [comment]
Do you think it will be add in darflow ?
Actually, I don't understand why yolov3 could not be supported directly in darkflow ?

**Comment by drew-neely** [comment]
Still not working

Layer [shortcut] not implemented

**Comment by JoffreyN** [comment]
Still not working...

Parsing ./cfg/yolo.cfg
Layer [shortcut] not implemented

**Comment by LucasMahieu** [comment]
Maybe the support of tiny yolo v3 would be a good beginning.
The shortcut layer is not used in Yolov3-tiny.

So, supporting yolov3 tiny is only question of upsampling layer (which can be implemented with a Resize Nearest Neighbor in tensorflow).

@ambr89 did you succeeded to run tiny yolo v3 with darkflow ?

**Comment by prdas31** [comment]
What about this project:
[https://github.com/qqwweee/keras-yolo3](url)

It uses Keras with TF backend. 
But not sure whether it can be used to train with my own classes.
Lots of work,, it seems :)

**Comment by shenyingying** [comment]
> Thanks for bringing this up.
> 
> I'll try to do minor refactoring gradually, so that the code is easier to read & extend for contributors. I admit after a while not looking at it I'm kind of lost.

@thtrieu  hi guys,have you refactoring it,does it work for yolov3,i am wondering for it,ths

**Comment by shenyingying** [comment]
> Here, someone has implemented yolov3 in tensorflow.
> https://github.com/mystic123/tensorflow-yolo-v3

@LucasMahieu  it's seems not work for AS demo

**Comment by sjain-stanford** [comment]
For those wanting to convert Yolo-v3 from Darknet to TensorFlow:

Yolo-v3 support was recently added to [DW2TF](https://github.com/jinyu121/DW2TF) (see [this PR](https://github.com/jinyu121/DW2TF/pull/4)).

Worth noting that unlike Darkflow which is also a runtime env for training/inference, DW2TF can only convert a Darknet model & weights to TensorFlow. Running training/inference on it would then be up to the user.

**Comment by willbattel** [comment]
Are people still trying to chip away at v3 support, or has this gone cold?

Last commit was 11 months ago 🤔

**Comment by azainab** [comment]
> @thtrieu

@thtrieu any update on this? is this issue resolved?

Layer [shortcut] not implemented

I still get the same error

**Comment by ambr89** [comment]
Hi everybody!

I've translate yolov3 from .ckpt to .pb and from .pb to .tflite 

I've use keras, something like this
`https://github.com/qqwweee/keras-yolo3`

**Comment by fadi212** [comment]
> Thanks for bringing this up.
> 
> I'll try to do minor refactoring gradually, so that the code is easier to read & extend for contributors. I admit after a while not looking at it I'm kind of lost.

Hi Sir,
Still waiting for that upadte. Have to use darkent and it is a bit difficult for windows users.

**Comment by fadi212** [comment]
> Hi everybody!
> 
> I've translate yolov3 from .ckpt to .pb and from .pb to .tflite
> 
> I've use keras, something like this
> `https://github.com/qqwweee/keras-yolo3`

Hi Sir,
Still waiting for that update. Have to use darknet and thats a bit difficult for windows users.

**Comment by JJeonggg** [comment]
Any update on this? still waiting for v3 support.

**Comment by chenyao96** [comment]
still waiting for v3 support
